src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -1798,16 +1870,37 @@ end = struct
             (Fn.compose add_endoscale_scalar_round
                (Endoscale_scalar_round.map ~f:reduce_to_v) )
     | Plonk_constraint.T (Lookup { w0; w1; w2; w3; w4; w5; w6 }) ->
+        (* table ID *)
+        let red_w0 = reduce_to_v w0 in
+        (* idx1 *)
+        let red_w1 = reduce_to_v w1 in
+        (* v1 *)
+        let red_w2 = reduce_to_v w2 in
+        (* idx2 *)
+        let red_w3 = reduce_to_v w3 in
+        (* v2 *)
+        let red_w4 = reduce_to_v w4 in
+        (* idx3 *)
+        let red_w5 = reduce_to_v w5 in
+        (* v3 *)
+        let red_w6 = reduce_to_v w6 in
         let vars =
-          [| Some (reduce_to_v w0)
-           ; Some (reduce_to_v w1)
-           ; Some (reduce_to_v w2)
-           ; Some (reduce_to_v w3)
-           ; Some (reduce_to_v w4)
-           ; Some (reduce_to_v w5)
-           ; Some (reduce_to_v w6)
+          [| Some red_w0
+           ; Some red_w1
+           ; Some red_w2
+           ; Some red_w3
+           ; Some red_w4
+           ; Some red_w5
+           ; Some red_w6
           |]
         in
+        let lookup1 = (red_w0, (red_w1, red_w2)) in
```

Comment:
Will these new changes allow the circuit writer to specify which table shall be used with a potential `Lookup` gate?

Reply:
Yes, it is the element `w0` of the record. That should be renamed.